python/ray/tests/test_state_api_log.py

```diff
@@ -999,9 +999,7 @@ def verify():
     with pytest.raises(requests.HTTPError) as e:
         list_logs(node_id=node_id)
 
-    assert (
-        f"Agent for node id: {node_id} doesn't exist." in e.value.response.json()["msg"]
-    )
+    assert e.value.response.status_code == 500
```

Comment:
This is a small behavior change: we no longer send tracebacks to users when there's an internal server error. Exposing tracebacks to users is a bad practice, as they should only be visible to developers and could pose security risks.  
By the way, I think the best status code here should be 404, since the node ID is not found, but I don't want to introduce API changes while working on the subprocess head module migration.